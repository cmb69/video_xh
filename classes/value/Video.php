<?php

/**
 * Copyright 2012-2023 Christoph M. Becker
 *
 * This file is part of Video_XH.
 *
 * Video_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Video_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Video_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Video\Value;

class Video
{
    /** @var array<string,string> */
    private $sources;

    /** @var string|null */
    private $poster;

    /** @var string|null */
    private $subtitle;

    /** @var int */
    private $date;

    /** @param array<string,string> $sources */
    public function __construct(array $sources, ?string $poster, ?string $subtitle, int $date)
    {
        $this->sources = $sources;
        $this->poster = $poster;
        $this->subtitle = $subtitle;
        $this->date = $date;
    }

    public function filename(): string
    {
        assert(!empty($this->sources));
        return key($this->sources);
    }

    /** @return array<string,string> */
    public function sources(): array
    {
        return $this->sources;
    }

    public function poster(): ?string
    {
        return $this->poster;
    }

    public function subtitle(): ?string
    {
        return $this->subtitle;
    }

    public function date(): int
    {
        return $this->date;
    }
}
