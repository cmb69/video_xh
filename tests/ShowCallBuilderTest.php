<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Video_XH.
 *
 * Video_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Video_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Video_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Video;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\View;
use Video\Model\VideoFinder;

class ShowCallBuilderTest extends TestCase
{
    public function testRendersCallBuilder(): void
    {
        $videoFinder = $this->createStub(VideoFinder::class);
        $videoFinder->method('availableVideos')->willReturn([]);
        $subject = new ShowCallBuilder(
            "./",
            XH_includeVar("./config/config.php", "plugin_cf")['video'],
            $videoFinder,
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")['video'])
        );

        $response = $subject();

        Approvals::verifyHtml($response->output());
    }
}
