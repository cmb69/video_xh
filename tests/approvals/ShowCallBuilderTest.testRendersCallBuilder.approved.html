
<script type="module" src="./video.min.js"></script>
<h1>Video â€“ Call Builder</h1>
<template id="video_call_builder">
  <form id="video_call_builder">
    <p>
      <label for="video_name">Video</label>
      <select id="video_name">
      </select>
    </p>
    <p>
      <label for="video_title">Title</label>
      <input id="video_title" type="text" value="">
    </p>
    <p>
      <label for="video_description">Description</label>
      <textarea id="video_description"></textarea>
    </p>
    <p>
      <label for="video_preload">Preload</label>
      <select id="video_preload">
        <option value="auto" selected>Auto</option>
        <option value="metadata" >Metadata</option>
        <option value="none" >None</option>
      </select>
    </p>
    <p>
      <label for="video_autoplay">Autoplay</label>
      <input id="video_autoplay" type="checkbox" >
    </p>
    <p>
      <label for="video_loop">Loop</label>
      <input id="video_loop" type="checkbox" >
    </p>
    <p>
      <label for="video_controls">Controls</label>
      <input id="video_controls" type="checkbox" checked>
    </p>
    <p>
      <label for="video_width">Width</label>
      <input id="video_width" type="text" value="512">
    </p>
    <p>
      <label for="video_height">Height</label>
      <input id="video_height" type="text" value="288">
    </p>
    <p>
      <label for="video_class">CSS Class</label>
      <input id="video_class" type="text" value="video_video">
    </p>
    <p>
      <textarea id="video_call"></textarea>
    </p>
  </form>
</template>
